### PR TITLE
Edit requirements section of connectors

### PR DESCRIPTION
**Pull request by rosewms**
Follow up from https://github.com/trinodb/trino/pull/8006

### CODE REVIEW

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 1]
```diff
@@ -8,12 +8,12 @@ The Cassandra connector allows querying data stored in
 Requirements
 ------------
 
-Requirements for using the connector in a catalog to connect to a Cassandra
-data source are:
+To connect to a Cassandra data source using the connector in a catalog, you
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/cassandra.rst:11]: What about `Connecting to a <XYZ> data source using the connector requires:`?

`connector in a catalog` is a bit confusing (specially that the reader hasn't been yet introduced to catalog <-> connector relation. The catalog and connector relationship is better suited in the Configuration section IMO.

File: docs/src/main/sphinx/connector/kafka.rst [hunk 2]
```diff
@@ -28,12 +28,11 @@ See the :doc:`kafka-tutorial`.
...
 ------------
 
-Requirements for using the connector in a catalog to connect to a Kafka
-data source are:
+To connect to a Kafka data source using the connector in a catalog, you need:
 
-* Kafka broker version is 0.10.0 or higher
-* Network access, by default on port 9092, from the Trino coordinator and
-  workers to the Kafka nodes.
+* Kafka broker version is 0.10.0 or higher.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/kafka.rst:33]: For consistency:

```suggestion
* Kafka broker version 0.10.0 or higher.
```

File: docs/src/main/sphinx/connector/mysql.rst [hunk 3]
```diff
@@ -9,12 +9,11 @@ systems like MySQL and Hive, or between two different MySQL instances.
...
 ------------
 
-Requirements for using the connector in a catalog to connect to a MySQL data
-source are:
+To connect to a MySQL data source using the connector in a catalog, you need:
 
-* MySQL 5.6, 5.7, 8.0 and higher
-* Network access, by default on port 3306, from the Trino coordinator and
-  workers to MySQL.
+* MySQL 5.7, 8.0 or higher.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/mysql.rst:14]: Why was `5.6` removed?

File: docs/src/main/sphinx/connector/sqlserver.rst [hunk 4]
```diff
@@ -10,12 +10,12 @@ between two different SQL Server instances.
...
 
-Requirements for using the connector in a catalog to connect to a SQL Server
-data source are:
+To connect to a SQL Server data source using the connector in a catalog, you
+need:
 
-* SQL Server 2012 or higher, or Azure SQL Database
-* Network access, by default on port 1433, from the Trino coordinator and
-  workers to SQL Server.
+* SQL Server 2012 or higher, or Azure SQL Database.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/sqlserver.rst:16]: Azure SQL Database is just hosted SQL Server IIRC. In same vein we could mention Google Cloud SQL or Amazon RDS too.

We should document this in some other way IMO. Maybe some text like `Hosted solutions should also work as long as the above requirements are met.`?

**Review by hashhar — COMMENTED** [review]
Thanks. Much better.

**Review by Ordinant — CHANGES_REQUESTED** [review]
One fix suggested. Otherwise, a masterwork.

File: docs/src/main/sphinx/connector/mysql.rst (hunk 3 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/mysql.rst:14]: Is it because `5.6` reached end-of-life in Feb 2021?

Comment (Ordinant) [reply docs/src/main/sphinx/connector/mysql.rst:14]: According to https://en.wikipedia.org/wiki/MySQL#Release_history, MySQL 5.6 is past its end-of-support date.

File: docs/src/main/sphinx/connector/oracle.rst [hunk 5]
```diff
@@ -9,12 +9,12 @@ like Oracle and Hive, or different Oracle database instances.
...
 
-Requirements for using the connector in a catalog to connect to a Oracle data
-source are:
+To connect to an Oracle data source using the connector in a catalog, you
+need:
 
-* Oracle 12 or higher
-* Network access, by default on port 1521, from the Trino coordinator and
-  workers to Oracle.
+* Oracle 12 or higher.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/oracle.rst:15]: remove trailing space

File: docs/src/main/sphinx/connector/oracle.rst [hunk 6]
```diff
@@ -9,12 +9,12 @@ like Oracle and Hive, or different Oracle database instances.
...
-Requirements for using the connector in a catalog to connect to a Oracle data
-source are:
+To connect to an Oracle data source using the connector in a catalog, you
+need:
 
-* Oracle 12 or higher
-* Network access, by default on port 1521, from the Trino coordinator and
-  workers to Oracle.
+* Oracle 12 or higher. 
+* Network access from the Trino coordinator and workers to Oracle.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/oracle.rst:16]: remove trailing space

File: docs/src/main/sphinx/connector/oracle.rst [hunk 7]
```diff
@@ -9,12 +9,12 @@ like Oracle and Hive, or different Oracle database instances.
...
-source are:
+To connect to an Oracle data source using the connector in a catalog, you
+need:
 
-* Oracle 12 or higher
-* Network access, by default on port 1521, from the Trino coordinator and
-  workers to Oracle.
+* Oracle 12 or higher. 
+* Network access from the Trino coordinator and workers to Oracle. 
+  Port 1521 is the default port and is used in the code examples.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/oracle.rst:17]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/cassandra.rst [hunk 8]
```diff
@@ -8,12 +8,12 @@ The Cassandra connector allows querying data stored in
...
+need:
 
-* Cassandra version 2.2 or higher
-* Network access, by default on port 9042, from the Trino coordinator and
-  workers to Cassandra.
+* Cassandra version 2.2 or higher.
+* Network access from the Trino coordinator and workers to Cassandra.
+  Port 9042 is the default port and is used in the code examples.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/cassandra.rst:16]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/kafka.rst [hunk 9]
```diff
@@ -28,12 +28,11 @@ See the :doc:`kafka-tutorial`.
...
+* Network access from the Trino coordinator and workers to the Kafka nodes.
+  Port 9092 is the default port and is used in the code examples.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/kafka.rst:35]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/mongodb.rst [hunk 10]
```diff
@@ -8,12 +8,11 @@ The ``mongodb`` connector allows the use of `MongoDB <https://www.mongodb.com/>`
...
-Requirements for using the connector in a catalog to connect to a MongoDB
-data source are:
+To connect to a MongoDB data source using the connector in a catalog, you need:
 
-* MongoDB 3.4 or higher
-* Network access, by default on port 27017, from the Trino coordinator and
-  workers to MongoDB.
+* MongoDB 3.4 or higher.
+* Network access from the Trino coordinator and workers to MongoDB.
+  Port 27017 is the default port and is used in the code examples.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/mongodb.rst:15]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/mysql.rst [hunk 11]
```diff
@@ -9,12 +9,11 @@ systems like MySQL and Hive, or between two different MySQL instances.
...
+* Network access from the Trino coordinator and workers to MySQL.
+  Port 3306 is the default port and is used in the code examples.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/mysql.rst:16]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/oracle.rst (hunk 7 above)
Comment (m57lyra) [inline docs/src/main/sphinx/connector/oracle.rst:17]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/postgresql.rst [hunk 12]
```diff
@@ -10,12 +10,12 @@ PostgreSQL instances.
...
-data source are:
+To connect to a PostgreSQL data source using the connector in a catalog, you
+need:
 
-* PostgreSQL 9.6 or higher
-* Network access, by default on port 5432, from the Trino coordinator and
-  workers to PostgreSQL.
+* PostgreSQL 9.6 or higher.
+* Network access from the Trino coordinator and workers to PostgreSQL.
+  Port 5432 is the default port and is used in the code examples.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/postgresql.rst:18]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/sqlserver.rst [hunk 13]
```diff
@@ -10,12 +10,12 @@ between two different SQL Server instances.
...
+* Network access from the Trino coordinator and workers to SQL Server.
+  Port 1433 is the default port and is used in the code examples.
```
Comment (m57lyra) [inline docs/src/main/sphinx/connector/sqlserver.rst:18]: Not sure why this appears to be indented

File: docs/src/main/sphinx/connector/oracle.rst (hunk 7 above)
Comment (Ordinant) [reply docs/src/main/sphinx/connector/oracle.rst:17]: Meant to be a continuation of the second bullet's paragraph, right?

File: docs/src/main/sphinx/connector/mysql.rst (hunk 3 above)
Comment (rosewms) [reply docs/src/main/sphinx/connector/mysql.rst:14]: They stopped supporting it in Feb 2021.

File: docs/src/main/sphinx/connector/oracle.rst (hunk 7 above)
Comment (rosewms) [reply docs/src/main/sphinx/connector/oracle.rst:17]: Correct, it is a continuation of the bullet above.

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/cassandra.rst:11]: We discussed this and want to leave it as is. The connection between catalog and connector is very important and should be here as well. A common misunderstanding is that you can configure a connector separately somehow, when in fact you can not ..

File: docs/src/main/sphinx/connector/cassandra.rst (hunk 8 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/cassandra.rst:16]: Here and all others ... this is correct as the sentence is part of the same bullet item.

File: docs/src/main/sphinx/connector/sqlserver.rst (hunk 4 above)
Comment (rosewms) [reply docs/src/main/sphinx/connector/sqlserver.rst:16]: we're putting this in future work, thanks.

**Review by m57lyra — COMMENTED** [review]
LGTM. Conforms to like passages in other connectors.

**Review by mosabua — APPROVED** [review]
Looks great now. Follow up work can add this to all the other connectors we are still missing ;-)

File: docs/src/main/sphinx/connector/oracle.rst [hunk 14]
```diff
@@ -9,12 +9,12 @@ like Oracle and Hive, or different Oracle database instances.
 Requirements
 ------------
 
-Requirements for using the connector in a catalog to connect to a Oracle data
-source are:
+To connect to an Oracle data source using the connector in a catalog, you
```
Comment (electrum) [inline docs/src/main/sphinx/connector/oracle.rst:12]: > To connect to Oracle using the connector, you need:

File: docs/src/main/sphinx/connector/oracle.rst [hunk 15]
```diff
@@ -276,7 +275,7 @@ Mapping character types
 ^^^^^^^^^^^^^^^^^^^^^^^
 
 Trino's ``VARCHAR(n)`` maps to ``VARCHAR2(n CHAR)`` if ``n`` is no greater than
-4000. A larger or unbounded ``VARCHAR`` maps to ``NCLOB``.
+1.    A larger or unbounded ``VARCHAR`` maps to ``NCLOB``.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/oracle.rst:278]: Revert. This was correct. Oracle has two VARCHAR types and we use decide which to use depending on length of the VARCHAR on our side - 4000 characters being the magic value.

**Review by hashhar — COMMENTED** [review]
LGTM % comment.

File: docs/src/main/sphinx/connector/oracle.rst (hunk 15 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/oracle.rst:278]: Just a line below this is also incorrect -

```
Trino's ``CHAR(n)`` maps to ``CHAR(n CHAR)`` if ``n`` is no greater than 2000.
```

should be changed to

```
Trino's ``CHAR(n)`` maps to ``CHAR(n CHAR)`` if ``n`` is no greater than 500.
```

Comment (rosewms) [reply docs/src/main/sphinx/connector/oracle.rst:278]: thanks @hashhar, good catch. I forgot that whenever you open the Oracle doc it makes that change, its some weird bug.

@mosabua, hashhar found a second error in the Oracle doc. Should I make the update with this PR or a separate one?

Comment (mosabua) [reply docs/src/main/sphinx/connector/oracle.rst:278]: Personally I would take it out from this PR and then send a second PR for the value changes. Whatever we do however .. lets get this PR merged asap @electrum

**Review by electrum — APPROVED** [review]

**Comment by electrum** [comment]
Thanks!

**Pull request merged**